### PR TITLE
[Merged by Bors] - feat: Add lemmas about finite products of measures

### PR DESCRIPTION
**Pull request by LorenzoLuccioli**
Add the following lemmas about finite product of measures and measurable equivalences:

- `piCongrLeft_apply_apply`
- `pi_map_piCongrLeft`
- `pi_map_piOptionEquivProd`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/MeasurableSpace/Basic.lean [hunk 1]
```diff
@@ -1762,6 +1762,11 @@ def piCongrLeft (f : δ ≃ δ') : (∀ b, π (f b)) ≃ᵐ ∀ a, π a := by
 theorem coe_piCongrLeft (f : δ ≃ δ') :
     ⇑(MeasurableEquiv.piCongrLeft π f) = f.piCongrLeft π := by rfl
 
+lemma piCongrLeft_apply_apply {ι ι' : Type*} (e : ι ≃ ι') {β : ι' → Type*}
+    [∀ i', MeasurableSpace (β i')] (x : (i : ι) → β (e i)) (i : ι) :
+    (piCongrLeft (fun i' ↦ β i') e) x (e i) = x i := by
```
Comment (RemyDegenne) [inline Mathlib/MeasureTheory/MeasurableSpace/Basic.lean:1767]:
```suggestion
    piCongrLeft (fun i' ↦ β i') e x (e i) = x i := by
```

File: Mathlib/MeasureTheory/Constructions/Pi.lean [hunk 2]
```diff
@@ -508,6 +508,46 @@ theorem ae_eq_set_pi {I : Set ι} {s t : ∀ i, Set (α i)} (h : ∀ i ∈ I, s
...
+  let e_meas : ((i : ι) → β (some i)) × β none ≃ᵐ ((i : Option ι) → β i) :=
+        MeasurableEquiv.piOptionEquivProd β |>.symm
+  have me := MeasurableEquiv.measurableEmbedding e_meas
+  have : e_meas ⁻¹' pi univ s
+      = (pi univ (fun i ↦ s (some i))) ×ˢ (s none) := by
+    ext x
+    simp only [mem_preimage, Set.mem_pi, mem_univ, forall_true_left, mem_prod]
+    constructor; tauto
+    intro h i
+    rcases i <;> tauto
```
Comment (RemyDegenne) [inline Mathlib/MeasureTheory/Constructions/Pi.lean:547]: There are several goals after that `constructor`, right?
```suggestion
    constructor
    · tauto
    · intro h i
      rcases i <;> tauto
```

File: Mathlib/MeasureTheory/Constructions/Pi.lean [hunk 3]
```diff
@@ -508,6 +508,46 @@ theorem ae_eq_set_pi {I : Set ι} {s t : ∀ i, Set (α i)} (h : ∀ i ∈ I, s
...
+  congr
+  all_goals rw [e.apply_symm_apply]
+
+lemma pi_map_piOptionEquivProd {β : Option ι → Type*} [∀ i, MeasurableSpace (β i)]
+    (μ : (i : Option ι) → Measure (β i)) [∀ (i : Option ι), SigmaFinite (μ i)] :
+    ((Measure.pi fun i ↦ μ (some i)).prod (μ none)).map
+      (MeasurableEquiv.piOptionEquivProd β).symm = Measure.pi μ := by
+  refine pi_eq (fun s _ ↦ ?_) |>.symm
+  let e_meas : ((i : ι) → β (some i)) × β none ≃ᵐ ((i : Option ι) → β i) :=
+        MeasurableEquiv.piOptionEquivProd β |>.symm
```
Comment (RemyDegenne) [inline Mathlib/MeasureTheory/Constructions/Pi.lean:539]:
```suggestion
  let e_meas : ((i : ι) → β (some i)) × β none ≃ᵐ ((i : Option ι) → β i) :=
    MeasurableEquiv.piOptionEquivProd β |>.symm
```

File: Mathlib/MeasureTheory/Constructions/Pi.lean [hunk 4]
```diff
@@ -508,6 +508,46 @@ theorem ae_eq_set_pi {I : Set ι} {s t : ∀ i, Set (α i)} (h : ∀ i ∈ I, s
   (ae_le_set_pi fun i hi => (h i hi).le).antisymm (ae_le_set_pi fun i hi => (h i hi).symm.le)
 #align measure_theory.measure.ae_eq_set_pi MeasureTheory.Measure.ae_eq_set_pi
 
+lemma pi_map_piCongrLeft [hι' : Fintype ι'] (e : ι ≃ ι') {β : ι' → Type*}
+    [∀ i, MeasurableSpace (β i)] (μ : (i : ι') → Measure (β i)) [∀ i, SigmaFinite (μ i)] :
+    (Measure.pi fun i ↦ μ (e i)).map (MeasurableEquiv.piCongrLeft (fun i ↦ β i) e)
+    = Measure.pi μ := by
```
Comment (RemyDegenne) [inline Mathlib/MeasureTheory/Constructions/Pi.lean:514]:
```suggestion
      = Measure.pi μ := by
```

**Comment by RemyDegenne** [comment]
Can you merge master into this PR? I think this will solve the filing CI check.

**Comment by RemyDegenne** [comment]
Thanks!
maintainer merge

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors r+

**Pull request closed without merging**